### PR TITLE
Add Accelerators section to Lightning docs

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Add `Accelerators` section to Lightning docs

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -42,6 +42,15 @@ PyTorch Lightning
    common/lightning_module
    common/trainer
 
+.. toctree::
+   :maxdepth: 1
+   :name: Accelerators
+   :caption: Accelerators
+
+   accelerators/gpu
+   accelerators/tpu
```
Comment (kaushikb11) [inline docs/source/index.rst:51]: The index could be further expanded to

* GPU
	* Single GPU Training
	* Multi GPU Training
	* Multi Node Training
* TPU
	* Single TPU Training
	* TPU Pod Training
....

Comment (tchaton) [reply docs/source/index.rst:51]: I like this.

**Review by tchaton — APPROVED** [review]
LGTM !

File: docs/source/accelerators/gpu.rst [hunk 2]
```diff
@@ -4,10 +4,29 @@
...
+-------------------
+
+Make sure you're running on a machine with at least one GPU. There's no need to specify any NVIDIA flags
+as Lightning will do it for you.
+
+.. testcode::
+    :skipif: torch.cuda.device_count() < 1
+
+    # train on 1 GPU (using dp mode)
+    trainer = Trainer(gpus=1)
```
Comment (rohitgr7) [inline docs/source/accelerators/gpu.rst:23]:
```suggestion
    # train on 1 GPU (using DP mode)
    trainer = Trainer(gpus=1)
    # or
    trainer = Trainer(devices=1, accelerator='gpu')
```
default is DP?

**Review by rohitgr7 — COMMENTED** [review]
build locally. still a few references for multi-gpu doc are there in codebase

File: docs/source/accelerators/gpu.rst (hunk 2 above)
Comment (kaushikb11) [reply docs/source/accelerators/gpu.rst]: Updated! And I don't think we should confuse users with new ways for now

**Comment by kaushikb11** [comment]
> still a few references for multi-gpu doc are there in codebase

I think I have updated all the references. Could you share the reference links?

**Comment by rohitgr7** [comment]
@kaushikb11 
https://github.com/PyTorchLightning/pytorch-lightning/blob/b57feccbff57271144dae70a82b1948f7a4cf7af/pytorch_lightning/core/lightning.py#L1056
few more in the same module.

also there are few more references of TPU here.
https://github.com/PyTorchLightning/pytorch-lightning/blame/docs/gpu_training/docs/source/accelerators/gpu.rst#L280

**Review by rohitgr7 — APPROVED** [review]

**Comment by awaelchli** [comment]
@kaushikb11 is there a notion page for the docs reorganization?

File: docs/source/index.rst (hunk 1 above)
Comment (kaushikb11) [reply docs/source/index.rst:51]: @awaelchli Here's the notion [page](https://www.notion.so/gridai/Lightning-docs-revamp-1853e181cf334c10a43ae963c5f1d33a). I will update #7666 and clean the notion page a bit more. 

The rationale for creating a distinct Accelerators section is to increase visibility and group them together for users to investigate. It was more difficult to look for different accelerators in the Common Cases section as they were listed in a different order and felt disorganized.

Comment (kaushikb11) [reply docs/source/index.rst:51]: The main priority right now would be sorting out the Getting Started section. It's a bit messy right now

```python
### Getting Started

1. Installation
2. Quick Start
    1. Intro 
    2. Prepare Data (Data Module) 
    3. Research Code (Lightning Module) 
    4. Abstract Boilerplate (Trainer) 
    5. Maximum Flexibility (Callbacks)
    6. Starter Templates
3.  From PyTorch to Lightning
```

Comment (kaushikb11) [reply docs/source/index.rst:51]: Created an issue https://github.com/PyTorchLightning/pytorch-lightning/issues/10765

**Review by Borda — APPROVED** [review]
